onnx/defs/schema.cc

```diff
@@ -107,30 +109,12 @@ OpSchemaRegistry* OpSchemaRegistry::Instance() {
 
 void OpSchema::CheckInputOutputType(struct InferenceContext& ctx) const {
   std::unordered_map<std::string, std::string> type_constraints;
-  if (inputs_.empty() && ctx.getNumInputs() > 0) {
-    fail_check(
-        "Node (",
-        domain(),
-        "::",
-        Name(),
-        ":",
-        since_version(),
-        ") takes zero inputs, but got ",
-        ctx.getNumInputs(),
-        " in graph");
-  }
-  if (outputs_.empty() && ctx.getNumOutputs() > 0) {
-    fail_check(
-        "Node (",
-        domain(),
-        "::",
-        Name(),
-        ":",
-        since_version(),
-        ") yields zero outputs, but got ",
-        ctx.getNumOutputs(),
-        " in graph");
-  }
+  // Check the number of inputs / output.
+  std::string node_info =
+      std::string("Node with schema(") + domain() + "::" + Name() + ":" + std::to_string(since_version()) + ")";
```

Comment:
minor nit: Better to turn this into a utility method `Opschema::FullName()` or some such name. That will avoid constructing this string, which will be unused most of the time.

Reply:
Hmmm ... I see it is being called from two places. (I guess that's why this check was missing in the first place ... there was an assumption that the model should have passed the basic checker before applying type/shape inference.) Anyway ... I guess we would have to have a slightly more complex mechanism to have a common error message. It may still be worth avoiding constructing error message strings until the error actually happens. You could pass a `const NodeProto* node` as a parameter (null when there is no node) instead of node_info.

Reply:
Hi, considering that the key difference between the two functions lies in whether `node name` is present, and our aim is to avoid meaningless constructions, I've added a common function that includes a conditional check for the existence of node_name. This is integrated into `fail_check`. I believe this approach will address these two concerns effectively. For a simpler implementation, use the pattern "Node(name) ..." regardless of whether the name exists. :D